### PR TITLE
Filter GitLab groups by prefix in GitlabOrgDiscoveryEntityProvider

### PR DESCRIPTION
**Pull request by matteosilv**
## Filter GitLab groups by prefix in GitlabOrgDiscoveryEntityProvider

- fixed invalid URL without scheme in produced annotations
- filter and compute the right dashed name for the imported group, by stripping the group path prefix if provided in configuration

### CODE REVIEW

**Comment by benjdlambert** [comment]
@jamieklassen do you want to take a look at this and see if it makes sense? I think that you might have created this?

**Comment by jamieklassen** [comment]
I didn't create the initial implementation, that was @dpfaffenbauer. However, this does a good job of covering the "group + descendants only" requirement which I described in #16025. I'm less confident when it comes to URLs in annotations -- I don't really know what those get used for.

**Comment by matteosilv** [comment]
Regarding the urls in annotations, I had to add the schemes by passing baseUrl instead of host, otherwise I was getting invalid url errors

**Comment by dpfaffenbauer** [comment]
@matteosilv You are using the optional baseUrl, which we don't. does the baseUrl always get set when not defined in config?

**Comment by matteosilv** [comment]
Yes its default value is ‘https://${host}’

**Comment by dpfaffenbauer** [comment]
than I guess its fine 💪  @matteosilv

**Comment by jamieklassen** [comment]
I learned that this PR fixes the behaviour of the edit button on the about card for these entities:

https://github.com/backstage/backstage/blob/e81995a8b65a6ce0d47812d978c5098a360feed6/plugins/catalog/src/components/AboutCard/AboutCard.tsx#L171-L179

which relies on the `backstage.io/edit-url` annotation being a valid URL:

https://github.com/backstage/backstage/blob/e81995a8b65a6ce0d47812d978c5098a360feed6/plugins/catalog/src/components/AboutCard/AboutCard.tsx#L105-L106

The code on master does not render clickable "Edit" buttons, whereas this PR does!

**Comment by matteosilv** [comment]
I added some documentation!

File: docs/integrations/gitlab/org.md [hunk 1]
```diff
@@ -31,4 +31,10 @@ catalog:
       yourProviderId:
         host: gitlab.com
         orgEnabled: true
+        group: org/teams # Optional. Must not end with slash. Accepts only groups under the provided path (excluded)
+        groupPattern: '[\s\S]*' # Optional. Filters found groups based on provided patter. Defaults to `[\s\S]*`, which means to not filter anything
```
Comment (jamieklassen) [inline docs/integrations/gitlab/org.md:35]:
```suggestion
        groupPattern: '[\s\S]*' # Optional. Filters found groups based on provided pattern. Defaults to `[\s\S]*`, which means to not filter anything
```

File: docs/integrations/gitlab/org.md [hunk 2]
```diff
@@ -31,4 +31,10 @@ catalog:
       yourProviderId:
         host: gitlab.com
         orgEnabled: true
+        group: org/teams # Optional. Must not end with slash. Accepts only groups under the provided path (excluded)
```
[the remark this answers is not in the document]
Comment (matteosilv) [reply docs/integrations/gitlab/org.md:34]: yes. It's perfect.

**Comment by matteosilv** [comment]
Hi guys, just wanted to refresh the thread to prevent autoclosing this. Next steps to merge? needs maintainer review?

**Review by Rugvip — APPROVED** [review]
Sorry, just swamped 😅 

Looks good, thank you! 🎉  Let's :shipit:

**Pull request merged**